Source/Core/DolphinWX/LuaScriptFrame.cpp

```diff
@@ -0,0 +1,553 @@
+
+  // Stop currently executing Lua thread
+  if (lua_thread != nullptr && lua_thread->IsRunning())
+  {
+    lua_thread->Kill();
+    lua_thread = nullptr;
+  }
+
+  main_frame->m_lua_script_frame = nullptr;
+
+  // Nullify GC manipulator function to prevent crash when lua console is closed
+  Movie::s_gc_manip_funcs[Movie::GCManipIndex::LuaGCManip] = nullptr;
+}
+
+//
+// CreateGUI
+//
+// Creates actual Lua console window.
+//
+void LuaScriptFrame::CreateGUI()
+{
+  this->SetSizeHints(wxDefaultSize, wxDefaultSize);
+
+  m_menubar = new wxMenuBar(0);
+  console_menu = new wxMenu();
+  clear =
+      new wxMenuItem(console_menu, wxID_ANY, wxString(_("Clear")), wxEmptyString, wxITEM_NORMAL);
+  console_menu->Append(clear);
+  help_menu = new wxMenu();
+  documentation = new wxMenuItem(help_menu, wxID_ANY, wxString(_("Lua Documentation")),
+                                 wxEmptyString, wxITEM_NORMAL);
+  api = new wxMenuItem(help_menu, wxID_ANY, wxString(_("Dolphin Lua API")), wxEmptyString,
+                       wxITEM_NORMAL);
+  help_menu->Append(documentation);
+  help_menu->Append(api);
+
+  m_menubar->Append(console_menu, _("Console"));
+  m_menubar->Append(help_menu, _("Help"));
+
+  this->SetMenuBar(m_menubar);
+
+  wxBoxSizer* main_sizer;
+  main_sizer = new wxBoxSizer(wxVERTICAL);
+
+  script_file_label =
+      new wxStaticText(this, wxID_ANY, _("Script File:"), wxDefaultPosition, wxDefaultSize, 0);
+  script_file_label->Wrap(-1);
+  main_sizer->Add(script_file_label, 0, wxALL, 5);
+
+  file_path = new wxTextCtrl(this, wxID_ANY, wxEmptyString, wxDefaultPosition, wxSize(300, -1), 0);
+  file_path->SetForegroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_WINDOWTEXT));
+
+  main_sizer->Add(file_path, 0, wxALL, 5);
+
+  wxBoxSizer* buttons;
+  buttons = new wxBoxSizer(wxHORIZONTAL);
+
+  browse_button = new wxButton(this, wxID_ANY, _("Browse..."), wxPoint(-1, -1), wxDefaultSize, 0);
+  buttons->Add(browse_button, 0, wxALL, 5);
+
+  run_button = new wxButton(this, wxID_ANY, _("Run"), wxDefaultPosition, wxDefaultSize, 0);
+  buttons->Add(run_button, 0, wxALL, 5);
+
+  stop_button = new wxButton(this, wxID_ANY, _("Stop"), wxDefaultPosition, wxDefaultSize, 0);
+  buttons->Add(stop_button, 0, wxALL, 5);
+
+  main_sizer->Add(buttons, 1, wxEXPAND, 5);
+
+  output_console_literal =
+      new wxStaticText(this, wxID_ANY, _("Output Console:"), wxDefaultPosition, wxDefaultSize, 0);
+  output_console_literal->Wrap(-1);
+  main_sizer->Add(output_console_literal, 0, wxALL, 5);
+
+  output_console = new wxTextCtrl(this, wxID_ANY, wxEmptyString, wxDefaultPosition,
+                                  wxSize(415, 200), wxHSCROLL | wxTE_MULTILINE | wxTE_READONLY);
+  main_sizer->Add(output_console, 0, wxALL, 6);
+
+  this->SetSizer(main_sizer);
+  this->Layout();
+
+  this->Centre(wxBOTH);
+
+  // Connect Events
+  this->Connect(clear->GetId(), wxEVT_COMMAND_MENU_SELECTED,

```

Comment:
wx's [Bind](http://docs.wxwidgets.org/3.0/classwx_evt_handler.html#a0f30c8fa5583b4a5f661897d63de3b62) should be utilized instead, as Connect is the older API for registering events.

e.g. The equivalent of this Connect call using Bind would be:

```cpp
Bind(wxEVT_MENU, &LuaScriptFrame::OnClearClicked, this, clear->GetId());
```